packages/web/server/api/edge-routers/concentrated-liquidity-router.ts

```diff
@@ -24,80 +14,9 @@ export const concentratedLiquidityRouter = createTRPCRouter({
         })
         .merge(UserOsmoAddressSchema.required())
     )
-    .query(async ({ input: { userOsmoAddress, sortDirection } }) => {
-      const result = await mapGetUserPositionDetails({
+    .query(({ input: { userOsmoAddress, sortDirection } }) =>
+      mapGetUserPositionDetails({
         userOsmoAddress,
-      });
-      return sort(result, "joinTime", sortDirection);
-    }),
-
-  getPositionPerformance: publicProcedure
-    .input(
-      z.object({
-        positionId: z.string(),
-      })
-    )
-    .query(async ({ input: { positionId } }) => {
-      const [position, performance] = await Promise.all([
-        queryCLPosition({ id: positionId }),
-        queryPositionPerformance({ positionId }),
-      ]);
-
-      const [
-        principalAssets,
-        currentPositionAssets,
-        unclaimedRewards,
-        totalEarned,
-      ] = await Promise.all([
-        mapAssetsToCoins({
-          rawAssets: performance.principal.assets,
-          calculatePrice: true,
-        }),
-        mapAssetsToCoins({
-          rawAssets: [position.position.asset0, position.position.asset1],
-          calculatePrice: true,
-        }),
-        getTotalClaimableRewards({
-          rawClaimableIncentiveRewards: position.position.claimable_incentives,
-          rawClaimableSpreadRewards: position.position.claimable_spread_rewards,
-          calculatePrice: true,
-        }),
-        getTotalEarned({
-          totalIncentivesRewards: performance.total_incentives_rewards,
-          totalSpreadRewards: performance.total_spread_rewards,
-        }),
-      ]);
-
-      const totalPrincipalValue = new PricePretty(
-        DEFAULT_VS_CURRENCY,
-        sum(principalAssets)
-      );
-      const currentPositionValue = new PricePretty(
-        DEFAULT_VS_CURRENCY,
-        sum(currentPositionAssets)
-      );
-      const unclaimedRewardsValue = new PricePretty(
-        DEFAULT_VS_CURRENCY,
-        sum(unclaimedRewards)
-      );
-      const totalEarnedValue = new PricePretty(
-        DEFAULT_VS_CURRENCY,
-        sum(totalEarned)
-      );
-
-      const roi = new RatePretty(
-        currentPositionValue
-          .toDec()
-          .add(unclaimedRewardsValue.toDec())
-          .add(totalEarnedValue.toDec())
-          .sub(totalPrincipalValue.toDec())
-          .quo(totalPrincipalValue.toDec())
-      );
-
-      return {
-        principalAssets,
-        totalEarned,
-        roi,
-      };
-    }),
```

Comment:
I believe we should leave position performance on a separate query. Maybe even separate the position details into more queries. It's nicer to see data load bit by bit rather than waiting for everything. Plus, if there's an error, it won't mess up the whole list like we've seen with Imperator

Reply:
I think combining all details into a single query greatly simplifies code complexity, especially on the client. Also, ROI calculation in the header of the position card needs the metrics in the expanded card, so there's not a clean break of collapsed vs expanded data.

To address the potential error issues with imperator (which I've brought up to their team), what are your thoughts on making the imperator portions optional and returning undefined if there's issues?

Reply:
Although it adds a bit more complexity, having a quicker initial load is worth it for users IMO. Splitting into more queries isn't too complicated, but we can handle that in a separate PR. For now, I agree that we shouldn't throw if Imperator fails, and just return `undefined` instead